Build issue with Perl 5.42 on Arch Linux

Hi, this follow https://github.com/LMS-Community/slimserver-vendor/issues/108

```
$ perl   --version

This is perl 5, version 42, subversion 0 (v5.42.0) built for x86_64-linux-thread-multi
```



```
build_module run tests:0 clean:1 hints 1 Audio-Scan-1.10 
tar zxf Audio-Scan-1.10.tar.gz
tar done
Checking if your kit is complete...
Looks good
Generating a Unix-style Makefile
Writing Makefile for Audio::Scan
Writing MYMETA.yml and MYMETA.json
Running Mkbootstrap for Scan ()
"/usr/bin/perl" "/usr/share/perl5/core_perl/ExtUtils/xsubpp"  -typemap '/usr/share/perl5/core_perl/ExtUtils/typemap'  Scan.xs > Scan.xsc
chmod 644 "Scan.bs"
"/usr/bin/perl" -MExtUtils::Command::MM -e 'cp_nonempty' -- Scan.bs blib/arch/auto/Audio/Scan/Scan.bs 644
cp lib/Audio/Scan.pm blib/lib/Audio/Scan.pm
Unparseable XSUB parameter: 'char *' in Scan.xs, line 220
Unparseable XSUB parameter: 'char *' in Scan.xs, line 272
Unparseable XSUB parameter: 'char *' in Scan.xs, line 288
Unparseable XSUB parameter: 'char *' in Scan.xs, line 312
Unparseable XSUB parameter: 'char *' in Scan.xs, line 328
Unparseable XSUB parameter: 'char *' in Scan.xs, line 366
Please specify prototyping behavior for Scan.xs (see perlxs manual)
make: *** [Makefile:363: Scan.c] Error 1
make failed, aborting
```

> 
> In the end I successfully ran buildme.sh with Perl 5.42 with the above changes to Audio-Scan-1.10/Scan.xs and replacing the `T_WATCHER` block in `EV-4.03/typemap` with:
> 
> ```
> T_WATCHER
>     if (!(SvROK($arg) && SvOBJECT(SvRV($arg)) && sv_derived_from($arg, \"EV::Watcher\")))
>         croak(\"$var is not of type EV::Watcher\");
>     $var = ($type)SvPVX(SvRV($arg));
> ```
> 
> I am super ignorant of xsubpp, this change was recommended by chatgpt. All my tests with Perl 5.42 have been successful thus far.

Thanks for this! I'm trying to restore LMS after a update which I should not have rushed into.

I have edited Scan.xs, which allows buildme.sh to run, but still have the test errors:

> libev/ev.c:1338:7: warning: ignoring return value of ‘read’ declared with attribute ‘warn_unused_result’ [-Wunused-result]
> 1338 |       read (evfd, &counter, sizeof (uint64_t));
>      |       ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
> libev/ev.c:1345:7: warning: ignoring return value of ‘read’ declared with attribute ‘warn_unused_result’ [-Wunused-result]
>  1345 |       read (evpipe [0], &dummy, 1);
>      |       ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
> make: *** [Makefile:352: EV.o] Error 1
> make test failed, aborting

I had edited `EV-4.03/typemap`  as you suggested, but after the failure, I notice that the edits are gone. Likewise, Scan.xs reverts to the original after a failed run of the script.

Thanks for any suggestions. I'm not as fluent on these build tools, esp. relating to perl.

--- a/Scan.xs
+++ b/Scan.xs
@@ -217,7 +217,7 @@ _generate_hash(const char *file)
 MODULE = Audio::Scan		PACKAGE = Audio::Scan
 
 HV *
-_scan( char *, char *suffix, PerlIO *infile, SV *path, int filter, int md5_size, int md5_offset )
+_scan( char *dummy, char *suffix, PerlIO *infile, SV *path, int filter, int md5_size, int md5_offset )
 CODE:
 {
   taghandler *hdl;
@@ -269,7 +269,7 @@ OUTPUT:
   RETVAL
   
 int
-_find_frame( char *, char *suffix, PerlIO *infile, SV *path, int offset )
+_find_frame( char *dummy, char *suffix, PerlIO *infile, SV *path, int offset )
 CODE:
 {
   taghandler *hdl;
@@ -285,7 +285,7 @@ OUTPUT:
   RETVAL
 
 HV *
-_find_frame_return_info( char *, char *suffix, PerlIO *infile, SV *path, int offset )
+_find_frame_return_info( char *dummy, char *suffix, PerlIO *infile, SV *path, int offset )
 CODE:
 {
   taghandler *hdl = _get_taghandler(suffix);
@@ -309,7 +309,7 @@ OUTPUT:
   RETVAL
 
 int
-is_supported(char *, SV *path)
+is_supported(char *dummy, SV *path)
 CODE:
 {
   char *suffix = strrchr( SvPVX(path), '.' );
@@ -325,7 +325,7 @@ OUTPUT:
   RETVAL
 
 SV *
-type_for(char *, SV *suffix)
+type_for(char *dummy, SV *suffix)
 CODE:
 {
   taghandler *hdl = NULL;
@@ -363,7 +363,7 @@ OUTPUT:
   RETVAL
 
 AV *
-extensions_for(char *, SV *type)
+extensions_for(char *dummy, SV *type)
 CODE:
 {
   int i, j;

